devDocs/startupShutdown.md

```diff
@@ -0,0 +1,179 @@
+    1. NVDA+q
+        - test: `startupShutdownNVDA.Quits from keyboard, Restarts`
+    1. An input gesture to restart
+    1. After changing some settings (eg installed add-ons or UI language), user prompted on dialog exit.
+    1. Via the NVDA menu -> Exit
+        - test: `startupShutdownNVDA.Quits from menu`
+1. A process sending `WM_QUIT`, eg a new NVDA process starting
+1. A handled crash (directly causes a new process to start, terminates unsafely)
+    - test: `startupShutdownNVDA.Restarts on crash, Restarts on braille crash`
+1. An unhandled crash (terminates unsafely)
+    - requires manual testing/confirmation
+1. An external command which kills the process (terminates unsafely) 
+1. Windows shutting down (terminates unsafely) (uses `wx.EVT_END_SESSION`)
+
+## Manual testing
+Instructions for testing startup / shutdown.
+
+### Start from shortcut
+Prerequisites:
+ - NVDA installed
+ - Shortcut enabled during installation
+
+Steps:
+ 1. Press (or emulate) Ctrl+Alt+N, observe NVDA starts up
+
+Variation:
+- At step 1. A version of NVDA is already running. Observe running version exists before installed version starts up.
+
+### Windows Sign-in screen, automatic start
+Prerequisites:
+ - NVDA installed
+ - Enable "Use NVDA during sign-in"
+
+Steps:
+ 1. Sign out (not lock) Windows
+ 1. Observe NVDA announces the Windows sign-in screen
+
+### UAC, automatic start
+Prerequisites:
+ - NVDA installed
+
+Steps:
+
+### Windows Successful sign-in, automatic start
+Prerequisites:
+ - NVDA installed
+ - Enable "Start NVDA after I sign in"
+
+Steps:
+ 1. Start Windows
+ 1. Sign in
+ 1. Observe NVDA starts
+
+### Running the *.exe
+
+Steps:
+ 1. Press `win+r`
+ 1. Enter <path to nvda.exe>
+ 1. Press enter
+ 1. Observe NVDA starts
+
+Variation:
+- using an installer (launcher)
+   -  eg: `C:\Users\username\Downloads\nvda_2021.1.exe`
+- using an installed copy
+   - just type `nvda` in place of the .exe
+- using a portable copy
+   - find and use the path to `nvda.exe`, located within the portable copy directory
+   - the installer allows you to create an installed copy and a portable copy
+
+### Running from source (runnvda.bat)
+Prerequisites
+- clone project and build NVDA (see [project readme](https://github.com/nvaccess/nvda/blob/master/readme.md#getting-the-source-code)).
+
+Steps:
+ 1. Run `runnvda.bat` from cmd
+ 1. Observe NVDA starts
+
+### An input gesture to restart
+
+Prerequisite:
+- Input gesture to restart is assigned to a key
+
+Steps:
+ 1. Press (or emulate) the input gesture
+ 1. Observe that NVDA exits
+ 1. Observe that a new instance is started
+
+## Technical notes
+
+These notes are aimed at developers, wishing to understand technical aspects of the NVDA start and exit.
+
+1. No more than one NVDA process instance should be running time. Interactions with itself could cause severe issues, some (non-exhaustive list) examples of sub-systems where this would be a problem:

```

Comment:
This seems incomplete and should probably say:
```suggestion
1. No more than one NVDA process instance should be running at the same time. Interactions with itself could cause severe issues, some (non-exhaustive list) examples of sub-systems where this would be a problem:
```

\or something similar